From 8d54833a0dcf91cb69b0cfb756d637a740708caa Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Wed, 12 Mar 2014 11:30:24 +0800
Subject: [PATCH 296/304] ENGR00232913 selinux:re-fix wifi tethering fail
 issue in selinux

Former commit only work in engineer build. Still have more denial info
in user build version.
In fact, just externd the file context will OK, for it will associate
the right domain to get the corresponding permissions.
In addition, delete obsoleted hostapd service in init.rc.

Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 evk_6sl/BoardConfig.mk              |  2 +-
 evk_6sl/sepolicy/file_contexts      |  2 ++
 evk_6sl/sepolicy/netd.te            |  3 ---
 imx6/etc/init.rc                    | 14 --------------
 sabreauto_6q/BoardConfig.mk         |  2 +-
 sabreauto_6q/sepolicy/file_contexts |  2 ++
 sabreauto_6q/sepolicy/netd.te       |  3 ---
 sabresd_6dq/BoardConfig.mk          |  2 +-
 sabresd_6dq/sepolicy/file_contexts  |  2 ++
 sabresd_6dq/sepolicy/netd.te        |  3 ---
 10 files changed, 9 insertions(+), 26 deletions(-)
 create mode 100644 evk_6sl/sepolicy/file_contexts
 delete mode 100644 evk_6sl/sepolicy/netd.te
 create mode 100644 sabreauto_6q/sepolicy/file_contexts
 delete mode 100644 sabreauto_6q/sepolicy/netd.te
 create mode 100644 sabresd_6dq/sepolicy/file_contexts
 delete mode 100644 sabresd_6dq/sepolicy/netd.te

diff --git a/evk_6sl/BoardConfig.mk b/evk_6sl/BoardConfig.mk
index eb80e5e..5849f83 100644
--- a/evk_6sl/BoardConfig.mk
+++ b/evk_6sl/BoardConfig.mk
@@ -83,4 +83,4 @@ BOARD_SEPOLICY_DIRS := \
        device/fsl/evk_6sl/sepolicy
 BOARD_SEPOLICY_UNION := \
        app.te \
-       netd.te
+       file_contexts
diff --git a/evk_6sl/sepolicy/file_contexts b/evk_6sl/sepolicy/file_contexts
new file mode 100644
index 0000000..ddb468a
--- /dev/null
+++ b/evk_6sl/sepolicy/file_contexts
@@ -0,0 +1,2 @@
+/system/bin/rtl_wpa_supplicant  u:object_r:wpa_exec:s0
+/system/bin/rtl_hostapd     u:object_r:hostapd_exec:s0
diff --git a/evk_6sl/sepolicy/netd.te b/evk_6sl/sepolicy/netd.te
deleted file mode 100644
index 80966f5..0000000
--- a/evk_6sl/sepolicy/netd.te
+++ /dev/null
@@ -1,3 +0,0 @@
-allow netd self:netlink_socket *;
-allow netd self:packet_socket *;
-allow netd unlabeled:sock_file { create_file_perms rwx_file_perms relabelfrom };
diff --git a/imx6/etc/init.rc b/imx6/etc/init.rc
index 055b0bd..95d7447 100755
--- a/imx6/etc/init.rc
+++ b/imx6/etc/init.rc
@@ -687,20 +687,6 @@ service iprenew_bt-pan /system/bin/dhcpcd -n
     disabled
     oneshot
 
-service hostapd /system/bin/hostapd /data/misc/wifi/hostapd.conf
-    socket hostapd_wlan0 dgram 660 root wifi
-    user root
-    group wifi
-    oneshot
-    disabled
-
-service rtl_hostapd /system/bin/hostapd /data/misc/wifi/hostapd.conf
-    socket hostapd_wlan0 dgram 660 root wifi
-    user root
-    group wifi
-    oneshot
-    disabled
-
 service ril-daemon /system/bin/rild
     class main
     socket rild stream 660 root radio
diff --git a/sabreauto_6q/BoardConfig.mk b/sabreauto_6q/BoardConfig.mk
index 9f2acd8..5d379e8 100644
--- a/sabreauto_6q/BoardConfig.mk
+++ b/sabreauto_6q/BoardConfig.mk
@@ -108,4 +108,4 @@ BOARD_SEPOLICY_DIRS := \
 
 BOARD_SEPOLICY_UNION := \
        app.te \
-       netd.te
+       file_contexts
diff --git a/sabreauto_6q/sepolicy/file_contexts b/sabreauto_6q/sepolicy/file_contexts
new file mode 100644
index 0000000..ddb468a
--- /dev/null
+++ b/sabreauto_6q/sepolicy/file_contexts
@@ -0,0 +1,2 @@
+/system/bin/rtl_wpa_supplicant  u:object_r:wpa_exec:s0
+/system/bin/rtl_hostapd     u:object_r:hostapd_exec:s0
diff --git a/sabreauto_6q/sepolicy/netd.te b/sabreauto_6q/sepolicy/netd.te
deleted file mode 100644
index 80966f5..0000000
--- a/sabreauto_6q/sepolicy/netd.te
+++ /dev/null
@@ -1,3 +0,0 @@
-allow netd self:netlink_socket *;
-allow netd self:packet_socket *;
-allow netd unlabeled:sock_file { create_file_perms rwx_file_perms relabelfrom };
diff --git a/sabresd_6dq/BoardConfig.mk b/sabresd_6dq/BoardConfig.mk
index 5c3245d..43d1c0f 100755
--- a/sabresd_6dq/BoardConfig.mk
+++ b/sabresd_6dq/BoardConfig.mk
@@ -114,4 +114,4 @@ BOARD_SEPOLICY_DIRS := \
 
 BOARD_SEPOLICY_UNION := \
        app.te \
-       netd.te
+       file_contexts
diff --git a/sabresd_6dq/sepolicy/file_contexts b/sabresd_6dq/sepolicy/file_contexts
new file mode 100644
index 0000000..ddb468a
--- /dev/null
+++ b/sabresd_6dq/sepolicy/file_contexts
@@ -0,0 +1,2 @@
+/system/bin/rtl_wpa_supplicant  u:object_r:wpa_exec:s0
+/system/bin/rtl_hostapd     u:object_r:hostapd_exec:s0
diff --git a/sabresd_6dq/sepolicy/netd.te b/sabresd_6dq/sepolicy/netd.te
deleted file mode 100644
index 80966f5..0000000
--- a/sabresd_6dq/sepolicy/netd.te
+++ /dev/null
@@ -1,3 +0,0 @@
-allow netd self:netlink_socket *;
-allow netd self:packet_socket *;
-allow netd unlabeled:sock_file { create_file_perms rwx_file_perms relabelfrom };
-- 
1.8.0

